src/vs/workbench/browser/parts/editor/breadcrumbsPicker.ts

```diff
@@ -440,7 +440,7 @@ export class BreadcrumbsOutlinePicker extends BreadcrumbsPicker {
 		this._symbolSortOrder = BreadcrumbsConfig.SymbolSortOrder.bindTo(this._configurationService);
 	}
 
-	protected _createTree(container: HTMLElement) {
+	protected _createTree(container: HTMLElement, element: BreadcrumbElement) {
```

Comment:
Not so nice - given that there is `_setInput`. Why not update/modify the filter in there

Reply:
I don't think that works.
I tried:
```
	protected _setInput(input: BreadcrumbElement): Promise<void> {
		const element = input as TreeElement;
		const model = OutlineModel.get(element)!;
		const textModel = model.textModel;
		this._overrideConfiguration = {
			resource: textModel.uri,
			overrideIdentifier: textModel.getLanguageIdentifier().language
		};
```
and
```
	private _getOutlineItemCompareType(): OutlineSortOrder {
		switch (this._symbolSortOrder.getValue(this._overrideConfiguration)) {
```

However in testing, getOutlineCompare gets called before _setInput, making the override undefined.

This makes sense as `_createTree` is called before `_setInput`, and `OutlineComparator` only takes a value rather than a function, so you have to have the override ready at the time of `_createTree`

Reply:
Just store the sorter in a property when _createTree is called and update it when _setInput is called

Reply:
I see what you mean, like this?